js/reference.js

```diff
@@ -1018,31 +1154,65 @@ var ERMrest = (function(module) {
          * @type {string}
          */
         get displayname() {
+            var self = this, table = this._pageRef._table, col;
             if (!this._displayname) {
-                var table = this._pageRef._table;
-                // if table has display row_name annotation
-                if (table.annotations.contains(module._annotations.TABLE_DISPLAY) &&
-                    table.annotations.get(module._annotations.TABLE_DISPLAY).contains("row_name")) {
-                    this._displayname = table.annotations.get(module._annotations.TABLE_DISPLAY).get("row_name"); // pattern
-
-                    // replace patterns with values
-                    var colnames = table.columns.names();
-                    for (var c = 0; c < colnames.length; c++) {
-                        var cname = colnames[c];
-                        var search = "{" + cname + "}";
-                        this._displayname = this._displayname.replace(new RegExp(search, 'g'), this._data[cname]);
-                    }
+                var annotation;
+                // If table has table-display annotation then set it in annotation variable
+                if (table.annotations.contains(module._annotations.TABLE_DISPLAY)) {
+                    annotation = module._getRecursiveAnnotationValue(module._contexts.ROWNAME, table.annotations.get(module._annotations.TABLE_DISPLAY).content);
+                }
 
+                // if annotation is populated and annotation has display.rowName property
+                if (annotation && typeof annotation.row_markdown_pattern === 'string') {
+                    var template = annotation.row_markdown_pattern;
+                    
+                    // Get formatted keyValues for a table for the data
+                    var keyValues = module._getFormattedKeyValues(this._pageRef, this._data);
+
+                    // get templated patten after replacing the values using Mustache
+                    var pattern = module._renderTemplate(template, keyValues);
+
+                    // Render markdown content for the pattern
+                    this._displayname = module._formatUtils.printMarkdown(pattern, { inline: true });
+                }
+                // no row_name annotation, use column with title, name, or id:text type
+                else if (typeof this._data.title === 'string') {
+                    col = table.columns.get("title");
```

Comment:
Is this operation case insensitive? If not, can you make it case-insensitive e.g both column names `Title` and `title` should work.